ray-operator/apis/ray/v1/raycluster_types.go

```diff
@@ -140,6 +140,14 @@ type RayClusterStatus struct {
 	Head HeadInfo `json:"head,omitempty"`
 	// Reason provides more information about current State
 	Reason string `json:"reason,omitempty"`
+
+	// Represents the latest available observations of a deployment's current state.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []RayClusterCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
```

Comment:
Why not use `metav1.Condition` instead? https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Condition

Reply:
I missed that. Thank you for reminding me.